columnflow/production/cms/pileup.py

```diff
@@ -18,13 +20,98 @@
 set_ak_column_f32 = functools.partial(set_ak_column, value_type=np.float32)
 
 
+logger = law.logger.get_logger(__name__)
+
+
 @producer(
     uses={"Pileup.nTrueInt"},
     produces={"pu_weight", "pu_weight_minbias_xs_up", "pu_weight_minbias_xs_down"},
     # only run on mc
     mc_only=True,
+    # function to determine the correction file
+    get_pileup_file=(lambda self, external_files: external_files.pu_sf),
 )
 def pu_weight(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+    """
+    Based on the number of primary vertices, assigns each event pileup weights using correctionlib.
+    """
+    # compute the indices for looking up weights
+    indices = events.Pileup.nTrueInt.to_numpy().astype("int32") - 1
+
+    # map the variable names from the corrector to our columns
+    variable_map = {
+        "NumTrueInteractions": indices,
+    }
+
+    for column_name, syst in (
+        ("pu_weight", "nominal"),
+        ("pu_weight_minbias_xs_up", "up"),
+        ("pu_weight_minbias_xs_down", "down"),
+    ):
+        # get the inputs for this type of variation
+        variable_map_syst = dict(variable_map, systematic=syst)
```

Comment:
```suggestion
        variable_map_syst = {**variable_map, "systematic": syst}
```

Feels a bit more verbose.